crates/api/src/gen.rs

```diff
@@ -0,0 +1,9363 @@
+    #[derive(Clone, Debug, Deserialize, Serialize)]
+    pub struct EventFilter {
+        #[serde(skip_serializing_if = "Option::is_none")]
+        #[serde(default)]
+        pub address: Option<Address>,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        #[serde(default)]
+        pub from_block: Option<BlockId>,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        #[serde(default)]
+        pub keys: Option<Vec<Vec<Felt>>>,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        #[serde(default)]
+        pub to_block: Option<BlockId>,
+    }
+
+    #[derive(Clone, Debug, Deserialize, Serialize)]
+    pub struct EventsChunk {
+        #[serde(skip_serializing_if = "Option::is_none")]
+        #[serde(default)]
+        pub continuation_token: Option<String>,
+        pub events: Vec<EmittedEvent>,
+    }
+
+    #[derive(Clone, Debug, Deserialize, Serialize)]
+    pub struct ExecutionResources {
+        #[serde(skip_serializing_if = "Option::is_none")]
+        #[serde(default)]
+        pub bitwise_builtin_applications: Option<i64>,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        #[serde(default)]
+        pub ec_op_builtin_applications: Option<i64>,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        #[serde(default)]
+        pub ecdsa_builtin_applications: Option<i64>,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        #[serde(default)]
+        pub keccak_builtin_applications: Option<i64>,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        #[serde(default)]
+        pub memory_holes: Option<i64>,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        #[serde(default)]
+        pub pedersen_builtin_applications: Option<i64>,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        #[serde(default)]
+        pub poseidon_builtin_applications: Option<i64>,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        #[serde(default)]
+        pub range_check_builtin_applications: Option<i64>,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        #[serde(default)]
+        pub segment_arena_builtin: Option<i64>,
+        pub steps: i64,
+    }
+
+    #[derive(Clone, Debug, Deserialize, Serialize)]
+    pub struct FeeEstimate {
+        pub gas_consumed: Felt,
+        pub gas_price: Felt,
+        pub overall_fee: Felt,
+        pub unit: PriceUnit,
+    }
+
+    #[derive(Clone, Debug, Deserialize, Serialize)]
+    pub struct FeePayment {
+        pub amount: Felt,
+        pub unit: PriceUnit,
+    }
+
+    #[derive(Clone, Debug, Deserialize, Serialize)]
+    #[serde(try_from = "String")]
+    pub struct Felt(String);
+
+    mod felt {
+        use super::jsonrpc;
+        use super::Felt;
+        use once_cell::sync::Lazy;
+        use regex::Regex;
+
+        static FELT_REGEX: Lazy<Regex> = Lazy::new(|| {
+            Regex::new("^0x(0|[a-fA-F1-9]{1}[a-fA-F0-9]{0,62})$").unwrap()
```

Comment:
An expect would be better here.

Reply:
Even though I don't thing `.expect()` makes a lot of difference compared to `.upwrap()`, I have also no problems with the change. Done.

Reply:
It's much better to know which regex failed to compile than just that the program panicked at the Nth line and to then have to dig in the Beerus code, especially generated code which tend to be less easy to digest.
This will also happen at runtime and will be user-facing in the context of Beerus. We shouldn't require our users to read our code.